linux/Jenkinsfile

```diff
@@ -1,126 +1,134 @@
+                                setup("${DISTRO}", "${ARCH}")
+                                unstash 'installercode'
+                                buildAndTest("${DISTRO}", "${ARCH}")
+                                if (params.uploadPackage.toBoolean()) {
+                                    echo "Upload artifacts for ${VERSION} - ${ARCH} - ${DISTRO}"
+                                    uploadArtifacts("${DISTRO}", "${ARCH}")
                                 }
                             }
+                        }
                     }
                 }
-            }
-        }
-        stage('Run non-x64 Deb packaging in parallel') {
-            when {
-                anyOf {
-                    expression { params.DISTRO_FILTER == 'all' }
-                    expression { params.DISTRO_FILTER == 'Debian' }
-                }
-            }
-            matrix {
-                tools {
-                    jdk "jdk-11.0.13+8"
-                }
-                when {
-                    anyOf {
-                        expression { params.VERSION_FILTER == 'all' }
-                        expression { params.VERSION_FILTER == env.VERSION }
-                    }
-                }
-                axes {
-                    axis {
-                        name 'ARCH'
-                        values 'aarch64', 'ppc64le', 's390x', 'armv7l'
-                    }
-                    axis {
-                        name 'VERSION'
-                        values '8', '11', '17', '18'
-                    }
-                }
-                excludes {
-                    exclude {
-                        axis {
-                            name 'ARCH'
-                            values 's390x'
-                        }
-                        axis {
-                            name 'VERSION'
-                            values '8'
+                stage('Build Installer for Redhat') {
+                    when  {
+                        beforeAgent true
+                        anyOf {
+                            expression{ params.DISTRO == 'all' }  // trigger debian build
+                            expression{ params.DISTRO == 'RedHat' }
                         }
                     }
-                }
-                agent {
-                    label "docker&&linux&&${ARCH}"
-                }
-                stages {
-                    stage('Build Installer') {
-                        steps {
-                            echo "Debian Installer Job for Temurin ${VERSION} - on ${ARCH}"
-                            setup()
+                    steps{
+                        dir('linuxRedHat') {
                             script {
-                                if (ARCH == "s390x") {
-                                    env.DOCKER_BUILDKIT=0
+                                DISTRO = "RedHat"
+                                echo "Installer Job for Temurin jdk ${VERSION} - ${ARCH} - ${DISTRO}"
```

Comment:
This seems to be repeating similar code to above, can it be extracted into a function?

Reply:
move into function` jenkinsStep()`